plugins/base/src/test/kotlin/pageMerger/PageNodeMergerTest.kt

```diff
@@ -187,4 +153,220 @@ class PageNodeMergerTest : BaseAbstractTest() {
             }
         }
     }
+
+    @RepeatedTest(3)
+    fun `should deterministically render same name property extensions`() {
```

Comment:
I covered all the cases I could think of, but if you can think of a corner case that isn't covered - I'd be happy to add tests for it